Can't start server after installing plugins

I have just set up etherpad lite on my VPS and everything worked fine out of the box. However, no matter what plugin I tried to install it seems to crash the server instantly after a restart. Uninstalling the plugin through `npm uninstall ` solves the problem, but doing this would mean that I can't use any plugins. I'm running Ubuntu 16.04 LTS, npm version 3.9.3, and nodejs version 6.2.1.

Here are the error messages:

> etherpad@ubuntu:~/etherpad-lite$ bin/run.sh 
> Ensure that all dependencies are up to date...  If this is the first time you have run Etherpad please be patient.
> Ensure jQuery is downloaded and up to date...
> Clearing minified cache...
> Ensure custom css/js files are created...
> Started Etherpad...
> [2016-06-11 06:50:05.992] [ERROR] console - (node:30075) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
> [2016-06-11 06:50:07.934] [INFO] console - Installed plugins: ep_cursortrace@2.0.15
> [2016-06-11 06:50:07.939] [INFO] console - Report bugs at https://github.com/ether/etherpad-lite/issues
> [2016-06-11 06:50:07.940] [INFO] console - Your Etherpad version is 1.6.0 (44370f1)
> [2016-06-11 06:50:08.068] [ERROR] console - Error: ENOENT: no such file or directory, stat '/home/etherpad/etherpad-lite/node_modules/ep_cursortrace/locales'
>     at Error (native)
>     at fs.statSync (fs.js:981:18)
>     at extractLangs (/home/etherpad/etherpad-lite/src/node/hooks/i18n.js:21:9)
>     at getAllLocales (/home/etherpad/etherpad-lite/src/node/hooks/i18n.js:44:34)
>     at Object.exports.expressCreateServer [as hook_fn] (/home/etherpad/etherpad-lite/src/node/hooks/i18n.js:83:17)
>     at normalizedhook (/home/etherpad/etherpad-lite/src/static/js/pluginfw/hooks.js:15:27)
>     at hookCallWrapper (/home/etherpad/etherpad-lite/src/static/js/pluginfw/hooks.js:21:14)
>     at /home/etherpad/etherpad-lite/src/static/js/pluginfw/hooks.js:76:14
>     at Function._.map._.collect (/home/etherpad/etherpad-lite/src/node_modules/underscore/underscore.js:172:24)
>     at Object.exports.callAll (/home/etherpad/etherpad-lite/src/static/js/pluginfw/hooks.js:75:24)
> [2016-06-11 06:50:08.068] [INFO] console - graceful shutdown...
> [2016-06-11 06:50:08.068] [INFO] console - db sucessfully closed.

Thanks for your time!

--- a/src/node/hooks/i18n.js
+++ b/src/node/hooks/i18n.js
@@ -5,7 +5,7 @@ var languages = require('languages4translatewiki')
   , npm = require('npm')
   , plugins = require('ep_etherpad-lite/static/js/pluginfw/plugins.js').plugins
   , semver = require('semver')
-  , existsSync = fs.statSync || fs.existsSync || path.existsSync
+  , existsSync = require('../utils/path_exists')
 ;
 
 

--- a/src/node/utils/caching_middleware.js
+++ b/src/node/utils/caching_middleware.js
@@ -21,8 +21,7 @@ var path = require('path');
 var zlib = require('zlib');
 var settings = require('./Settings');
 var semver = require('semver');
-
-var existsSync = fs.statSync || fs.existsSync || path.existsSync;
+var existsSync = require('./path_exists');
 
 var CACHE_DIR = path.normalize(path.join(settings.root, 'var/'));
 CACHE_DIR = existsSync(CACHE_DIR) ? CACHE_DIR : undefined;

--- a/src/node/utils/path_exists.js
+++ b/src/node/utils/path_exists.js
@@ -1,0 +1,15 @@
+var fs = require('fs');
+
+var check = function(path) {
+  var existsSync = fs.statSync || fs.existsSync || path.existsSync;
+
+  var result;
+  try {
+    result = existsSync(path);
+  } catch (e) {
+    result = false;
+  }
+  return result;
+}
+
+module.exports = check;

